tinygrad/tensor.py

```diff
@@ -211,6 +211,10 @@ def full_like(self, fill_value:Scalar, **kwargs):
   def zeros_like(self, **kwargs): return self.full_like(0, **kwargs)
   def ones_like(self, **kwargs): return self.full_like(1, **kwargs)
 
+  def one_hot(self, num_classes:int=-1, **kwargs) -> Tensor:
+    num_classes = self.max().item()+1 if num_classes == -1 else num_classes
```

Comment:
can you do this without going through item()? otherwise it's value dependent and not jittable. it's fine to always require a num_classes i think (see how jax does it)

also can you move this a suitable section under `functional nn ops` now it's not a creation function? same for test

Reply:
yeah, makes sense my mistake, I looked at Jax and as you thought they have num_classes as required argument.